crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -23,40 +20,84 @@ impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
+
+        let (leading_definition_comments, trailing_definition_comments) =
+            dangling_comments.split_at(trailing_definition_comments_start);
+
+        if let Some(last_decorator) = decorator_list.last() {
+            f.join_with(hard_line_break())
+                .entries(decorator_list.iter().formatted())
+                .finish()?;
 
-        if !decorator_list.is_empty() {
-            hard_line_break().fmt(f)?;
+            if leading_definition_comments.is_empty() {
+                write!(f, [hard_line_break()])?;
+            } else {
+                // Write any leading definition comments (between last decorator and the header)
+                // while maintaining the right amount of empty lines between the comment
+                // and the last decorator.
+                let decorator_end =
+                    skip_trailing_trivia(last_decorator.end(), f.context().source());
+
+                let leading_line = if lines_after(decorator_end, f.context().source()) <= 1 {
+                    hard_line_break()
+                } else {
+                    empty_line()
+                };
+
+                write!(
+                    f,
+                    [leading_line, leading_comments(leading_definition_comments)]
+                )?;
+            }
         }
 
         write!(f, [text("class"), space(), name.format()])?;
 
-        if arguments
-            .as_ref()
-            .is_some_and(|Arguments { args, keywords, .. }| {
-                !(args.is_empty() && keywords.is_empty())
-            })
-        {
-            parenthesized(
-                "(",
-                &FormatInheritanceClause {
-                    class_definition: item,
-                },
-                ")",
-            )
-            .fmt(f)?;
+        if let Some(arguments) = arguments {
+            // Drop empty parentheses, e.g., in:
+            // ```python
+            // class A():
+            //     ...
+            // ```
+            //
+            // However, preserve any dangling end-of-line comments, e.g., in:
+            // ```python
+            // class A(  # comment
+            // ):
+            //     ...
+            //
+            // If the arguments contain any dangling own-line comments, we retain the parentheses,
+            // e.g., in:
+            // ```python
+            // class A(  # comment
+            //     # comment
+            // ):
+            //     ...
+            // ```
+            if arguments.args.is_empty()
+                && arguments.keywords.is_empty()
+                && comments
+                    .dangling_comments(arguments)
+                    .iter()
+                    .all(|comment| comment.line_position().is_end_of_line())
+            {
+                let dangling = comments.dangling_comments(arguments);
+                write!(f, [trailing_comments(dangling)])?;
```

Comment:
I think this is the first usage where another node formats a node's dangling comments. I also don't understand the reason for it from the comment above because all examples in the comment above have parentheses, but the point here seems to be to remove the parentheses. 

Why not move this into `Arguments`? It removes the need for querying the dangling comments here (we only have to do it in arguments) and is more in line of what we do elsewhere.

Reply:
I could handle it in placement instead of here?

Reply:
That might be easier to understand if it is easy enough